flake.nix

```diff
@@ -46,18 +46,11 @@
             configureFlags = [ "-native-compiler" "yes" ];
           });
         })
+        self.overlays.default
```

Comment:
not the biggest fan of applying the own overlay when declaring `pkgs`. can we do it another way? maybe a common function that both the overlays and `outputs` call.